File: SistemaLocacaoVeiculosPHP/model/Locacao.php
<?php

include_once "Automovel.php";

include_once "Cliente.php";

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of Locacao
 *
 * @author feijo
 */
class Locacao {
    private $id_locacao;
    private $dt_locacao;
    private $hora_locacao;
    private $dt_devolucao;
    private $hora_devolucao;
    private $km;
    private $vl_calcao;
    private $vl_locacao;
    private $devolvido;
    private $situacao;
    private $automovel;
    private $cliente;
    
    function __construct($cpf,$renavan) {
        $marca = new Marca();
        $modelo = new Modelo($marca);
        $this->automovel = new Automovel($modelo->getDescricao());
        $this->automovel->setRenavan($renavan);
        $this->cliente = new Cliente();
        $this->cliente->setCpf_cli($cpf);
        
       
    }

    public function getId_locacao() {
        return $this->id_locacao;
    }

    public function getDt_locacao() {
        return $this->dt_locacao;
    }

    public function getHr_locacao() {
        return $this->hora_locacao;
    }

    public function getDt_devolucao() {
        return $this->dt_devolucao;
    }

    public function getHr_devolucao() {
        return $this->hora_devolucao;
    }

    public function getKm() {
        return $this->km;
    }

    public function getVl_calcao() {
        return $this->vl_calcao;
    }

    public function getVl_locacao() {
        return $this->vl_locacao;
    }

    public function getDevolvido() {
        return $this->devolvido;
    }

    public function getCliente() {
        return $this->cliente;
    }

    public function getAutomovel() {
        return $this->automovel;
    }

    public function getSituacao() {
        return $this->situacao;
    }

    public function setId_locacao($id_locacao) {
        $this->id_locacao = $id_locacao;
    }

    public function setDt_locacao($dt_locacao) {
        $this->dt_locacao = $dt_locacao;
    }

    public function setHora_locacao($hora_locacao) {
        $this->hora_locacao = $hora_locacao;
    }

    public function setDt_devolucao($dt_devolucao) {
        $this->dt_devolucao = $dt_devolucao;
    }

    public function setHora_devolucao($hora_devolucao) {
        $this->hora_devolucao = $hora_devolucao;
    }

    public function setKm($km) {
        $this->km = $km;
    }

    public function setVl_calcao($vl_calcao) {
        $this->vl_calcao = $vl_calcao;
    }

    public function setVl_locacao($vl_locacao) {
        $this->vl_locacao = $vl_locacao;
    }

    public function setDevolvido($devolvido) {
        $this->devolvido = $devolvido;
    }

    public function setCliente($cliente) {
        $this->cliente = $cliente;
    }

    public function setAutomovel($automovel) {
        $this->automovel = $automovel;
    }

    public function setSituacao($situacao) {
        $this->situacao = $situacao;
    }

        
    function __toString() {
        return "ID = $this->id_locacao SITUACAO = $this->situacao CLIENTE =".var_dump($this->cliente)." DEVOLVIDO  = $this->devolvido DATA DEVOLUCAO  = $this->dt_devolucao DATA LOCACAO  = $this->dt_locacao HORA DEVOLUCAO = $this->hora_devolucao HORA LOCACAO = $this->hora_locacao KILOMETRAGEM $this->km.KM VALOR CALCAO = $this->vl_calcao VALOR LOCACAO = $this->vl_locacao AUTOMOVEL =  $this->automovel";
    }


}

File: SistemaLocacaoVeiculosJAVA/src/model/Locacao.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

import java.sql.Date;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Time;

/**
 *
 * @author feijo
 */
public class Locacao {
    
    private String id_locacao;
    private Date dtLocacao;
    private Date dtDevolucao;
    private Time hrLocacao;
    private Time hrDevolucao;
    private Long km;
    private double vlCaucao;
    private double vlLocacao;
    private Cliente cliente = new Cliente();
    private Automovel automovel = new Automovel();
    private boolean situacao;
    private boolean devolvido;

    public Locacao(Cliente cliente, Automovel automovel) {
        this.cliente = cliente;
        this.automovel = automovel;
    }

    public Locacao(ResultSet rs) throws SQLException {
        this.id_locacao = rs.getString("id_locacao");
        this.dtLocacao = rs.getDate("dt_locacao");
        this.dtDevolucao = rs.getDate("dt_devolucao");
        this.hrLocacao = rs.getTime("hr_locacao");
        this.hrDevolucao = rs.getTime("hr_devolucao");
        this.km = rs.getLong("km");
        this.vlCaucao = rs.getDouble("valor_caucao");
        this.vlLocacao = rs.getDouble("valor_locacao");
        this.cliente.setCpf(rs.getLong("cpf_cli"));
        this.automovel.setRenavan(rs.getLong("renavan"));
        this.situacao = rs.getBoolean("situacao");
        this.devolvido = rs.getBoolean("devolvido");
    }
    

    public Locacao(Date dtLocacao, Date dtDevolucao, Time hrLocacao, Time hrDevolucao, Long km, double vlCaucao, double vlLocacao, Boolean devolvido, Cliente cliente, Automovel automovel, boolean situacao) {
        this.dtLocacao = dtLocacao;
        this.dtDevolucao = dtDevolucao;
        this.hrLocacao = hrLocacao;
        this.hrDevolucao = hrDevolucao;
        this.km = km;
        this.vlCaucao = vlCaucao;
        this.vlLocacao = vlLocacao;
        this.devolvido = devolvido;
        this.cliente = cliente;
        this.automovel = automovel;
        this.situacao = situacao;
    }

    public String getId_locacao() {
        return id_locacao;
    }

    public void setId_locacao(String id_locacao) {
        this.id_locacao = id_locacao;
    }

    public Date getDtLocacao() {
        return dtLocacao;
    }

    public void setDtLocacao(Date dtLocacao) {
        this.dtLocacao = dtLocacao;
    }

    public Date getDtDevolucao() {
        return dtDevolucao;
    }

    public void setDtDevolucao(Date dtDevolucao) {
        this.dtDevolucao = dtDevolucao;
    }

    public Time getHrLocacao() {
        return hrLocacao;
    }

    public void setHrLocacao(Time hrLocacao) {
        this.hrLocacao = hrLocacao;
    }

    public Time getHrDevolucao() {
        return hrDevolucao;
    }

    public void setHrDevolucao(Time hrDevolucao) {
        this.hrDevolucao = hrDevolucao;
    }

    public Long getKm() {
        return km;
    }

    public void setKm(Long km) {
        this.km = km;
    }

    public double getVlCaucao() {
        return vlCaucao;
    }

    public void setVlCaucao(double vlCaucao) {
        this.vlCaucao = vlCaucao;
    }

    public double getVlLocacao() {
        return vlLocacao;
    }

    public void setVlLocacao(double vlLocacao) {
        this.vlLocacao = vlLocacao;
    }

    public Boolean getDevolvido() {
        return devolvido;
    }

    public void setDevolvido(Boolean devolvido) {
        this.devolvido = devolvido;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public Automovel getAutomovel() {
        return automovel;
    }

    public void setAutomovel(Automovel automovel) {
        this.automovel = automovel;
    }

    public boolean getSituacao() {
        return situacao;
    }

    public void setSituacao(boolean situacao) {
        this.situacao = situacao;
    }
    @Override
    public String toString() {
        return "Locacao{" + "\n\tdtLocacao = " + dtLocacao + "\n\tdtDevolucao = " + dtDevolucao + "\n\thrLocacao = " + hrLocacao + "\n\thrDevolucao = " + hrDevolucao + "\n\tkm = " + km + "\n\tvlCaucao = " + vlCaucao + "\n\tvlLocacao = " + vlLocacao + "\n\tDevolvido = " + devolvido + "\n\tCliente = " + cliente + "\n\tAutomovel = " + automovel + "\n\tsituacao = " + situacao + "\n}";
    }
}
